docs/doxygen/structure_review.md

```diff
@@ -1,42 +1,54 @@
+2. Currently, IDGPredict is implemented using a custom interface. We should factor out this code and convert it to the generic DPStep interface, so it can also be used outside of DDECal.
+3. Refactor the nested DDECal model pipeline so that it can be defined separately for every direction instead of one to rule them all. The metadata must not change in any of the nested model steps of DDECal. This requires refactoring the steps setup in the constructor of DDECal.cc.
+4. Remove the model column from the buffer and create a ColumnReader that can read model columns that can be used in DDECal.
 
+## Suggested new parset for DDECal
 From a user perspective we propose the following additional parset definitions for DDECal:
-- `ddecal.modelnextsteps`: Defines the default steps for the models that need to be applied to every direction before applying DDECal. Each step can have its own parset entries.
-- `ddecal.modelnextsteps.<direction>`: Allows steps to be defined for a specific direction. Overrides the default steps for this direction.
+- `ddecal.modelnextsteps`: Defines the default steps for the models that need to be applied to every direction before using the output for calibration. Each step can have its own parset entries.
+- `ddecal.modelnextsteps.<direction>`: Allows steps to be defined for a specific direction. Overrides the default steps for this direction. `<direction>` is the direction name, which is the patch name in the `ddecal.sourcedb`, the region label in the `ddecal.regions`, or the model column name.
+- `ddecal.modeldatacolumns`: (optional) Names of the columns that are to be used in DDECal as models.
 
 An example of a parset for DDECal using different models for different directions is as follows:
 ```
 ddecal.images = [image1.fits, image2.fits]  
-ddecal.regions = ds9.reg  
-ddecal.sourcedb = ATeam.sourcedb  
+ddecal.regions = ds9.reg         # Contains regions names Region_1, Region_2
+ddecal.sourcedb = ATeam.sourcedb # Contains patches names CasA, CygA, and PerA (last one is ignored)
+ddecal.modeldatacolumns = [MODEL_DATA_1]
 ddecal.directions = [CasA, CygA] # subselection of patches from sourcedb
 
-ddecal.modelnextsteps = [applycal3]  
-ddecal.modelnextsteps.CasA = [applycal4, uvwflag3, applybeam3]  
-ddecal.modelnextsteps.CygA = [applycal5, applybeam4]  
+ddecal.modelnextsteps = [applycal]
+ddecal.modelnextsteps.CasA = [applycal, uvwflag, applybeam]  # Putting a UVWFlag here means applying a UV cut for this direction
+ddecal.modelnextsteps.CygA = [applycal, applybeam]
+ddecal.modelnextsteps.MODEL_DATA_1 = []
 
-applycal5.h5parm = known_corruptions.h5 # Will automagically get right direction
+applycal.h5parm = known_corruptions.h5 # Will automagically get right direction, i.e. the model data in direction CasA will be corrupted with the solution in direction 'CasA'
```

Comment:
corrupted? That doesn't sound good... Maybe say "The solution in direction x will be applied to the model data that has the same direction."

Reply:
It's jargon, probably users would understand. 'Correct' means 'correct for the instrumental error', whereas 'corrupt' means 'apply the instrumental error to data'.